waspc/data/Generator/templates/sdk/wasp/auth/user.ts

```diff
@@ -26,11 +30,22 @@ export function getFirstProviderUserId(user?: UserEntityWithAuth): string | null
   return user.auth.identities[0].providerUserId ?? null;
 }
 
+// PRIVATE API
+export function enrichAuthUser(data: AuthUserData): AuthUser {
+  return {
+    ...data,
+    getFirstProviderUserId: () => {
```

Comment:
Again, not related to this PR, but I forgot...

How comes this is a method instead of a property: `user.firstProviderUserId`?

Especially considering this is something we can precompute for each because it never changes... Or does it?

Reply:
It never changes, but it felt like more logic heavy piece on the `user` so I wanted to use a method instead of a getter 🤷

Reply:
Got it, but what about this?
> Especially considering this is something we can precompute for each because it never changes... Or does it?

It's logic heavy because we compute it every time. But if it never changes, why not compute it once (during the construction of AuthUser), and then just use a property?

I'm not sure I prefer one solution over the other, just putting it out there.

Reply:
It feels weird to me to call a property `firstProviderUserId` since it _feels_ dynamic to me. I could precompute it and store it some closure variable. But still, it's super cheap to compute it so I'd keep it simple for now.